pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -442,6 +443,9 @@ func (s *Scheduler) add(ctx context.Context, pod *corev1.Pod) error {
 	if err := s.addToInflightNode(ctx, pod); err == nil {
 		return nil
 	}
+	if len(s.nodeClaimTemplates) == 0 {
+		return serrors.Wrap(fmt.Errorf("nodepool requirements filtered out all compatible instance types"), "Pod", klog.KRef("", pod.Name))
```

Comment:
Should this be "nodepool requirements filtered out all available instance types"? Compatible ITs in my opinion are the ones that we get after filtering by requirements.